packages/fxa-settings/src/pages/Signin/SigninTotpCode/container.tsx

```diff
@@ -0,0 +1,156 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useNavigate, useLocation } from '@reach/router';
+import { useValidatedQueryParams } from '../../../lib/hooks/useValidate';
+import { SigninQueryParams } from '../../../models/pages/signin';
+import { SigninTotpCode } from './index';
+import { gql, useMutation } from '@apollo/client';
+import GleanMetrics from '../../../lib/glean';
+import { logViewEvent } from '../../../lib/metrics';
+import { hardNavigate } from 'fxa-react/lib/utils';
+import {
+  AuthUiError,
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from '../../../lib/auth-errors/auth-errors';
+import { currentAccount, sessionToken } from '../../../lib/cache';
+import { MozServices } from '../../../lib/types';
+import VerificationMethods from '../../../constants/verification-methods';
+import VerificationReasons from '../../../constants/verification-reasons';
+import { GraphQLError } from 'graphql';
+
+export const viewName = 'signin-totp-code';
+
+export type SigninTotpCodeContainerProps = {
+  serviceName: MozServices;
+};
+
+export const SigninTotpCodeContainer = ({
+  serviceName,
+}: SigninTotpCodeContainerProps & RouteComponentProps) => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const storedLocalAccount = currentAccount();
+
+  // TBD: A bit of a work around. In the previous signup step
+  // a new session token should have been generated. Setting
+  // it here to ensure that the authorization header is set
+  // when it comes time to verify the totp code.
+  if (storedLocalAccount?.sessionToken) {
+    sessionToken(storedLocalAccount?.sessionToken);
+  }
+
+  const { queryParamModel } = useValidatedQueryParams(SigninQueryParams);
+  const [verifyTotpCode] = useMutation(gql`
+    mutation VerifyTotp($input: VerifyTotpInput!) {
+      verifyTotp(input: $input) {
+        success
+      }
+    }
+  `);
+
+  const handleNavigation = () => {
```

Comment:
We've mostly had functions with API calls happen in the container component. I'm OK with this happening here, but it makes me wonder if we should move `navigationHandler` out of the Signin page component for consistency... I'm curious why you felt it was better here, maybe because it references the query param model?

Reply:
I put it here cause it's not part of the presentation layer, and it the query param model was already present here.